### DISCUSSION TITLE
"Needs editing" and source string changes clarification

### DISCUSSION THREAD

**Discussion by hicksjacobp** [WeblateOrg/weblate#5335] [Q&A]
I'm looking for some clarification for how translation strings may or may not automagically determine that a translation string needs changes based on the source string changes.

Per this Weblate documentation: <https://docs.weblate.org/en/latest/faq.html#why-does-weblate-still-show-old-translation-strings-when-i-ve-updated-the-template>

> Weblate does not try to manipulate the translation files in any way other than allowing translators to translate. So it also does not update the translatable files when the template or source code have been changed. You simply have to do this manually and push changes to the repository, Weblate will then pick up the changes automatically.

The tooltip on the "Needs editing" checkbox for a given string states:

> Strings are usually marked as "Needs editing" after the source string is updated, or when marked as such manually.

So is this all just saying that Weblate will not automagically change a translation based on a source string change, but it will automagically set the "Needs editing" flag when the source string changes?

**Comment by nijel** [comment]
Yes, that sounds correct.

The documentation talks about translation files and those are not affected at all by the "Needs editing" being set in Weblate.

**Comment by poVoq** [comment]
Is there a way to disable this automatic "needs editing" check?

**Reply by nijel** [reply]
No, this is currently not configurable:

https://github.com/WeblateOrg/weblate/blob/48498e5560d86b4f79c5a68ee654407750080234/weblate/trans/models/unit.py#L916-L945

**Reply by nijel** [reply]
@poVoq  If you want to make it configurable, please describe the behavior and use case in https://github.com/WeblateOrg/weblate/issues/16177.
